api/preprints/serializers.py

```diff
@@ -79,6 +80,7 @@ class PreprintSerializer(JSONAPISerializer):
     title = ser.CharField(source='node.title', required=False)
     description = ser.CharField(required=False, allow_blank=True, allow_null=True, source='node.description')
     tags = JSONAPIListField(child=NodeTagField(), required=False, source='node.tags')
+    node_is_public = ser.BooleanField(read_only=True, source='node__is_public')
```

Comment:
If there's a way to make that work, that'd be lovely, but I get a 400:
```json
{
    "errors": [
        {
            "source": {
                "parameter": "filter"
            },
            "meta": {},
            "detail": "'node__is_public' is not a valid field for this endpoint."
        }
    ]
}
```
even if I add `node__is_public` to `filterable_fields`

Reply:
I was about to be very surprised and pleased and possibly a little concerned if that worked.

Reply:
I don't think it would take a *lot* to update the filtering to handle this; it's probably just the verification that a field is named what you think it is, but it would be a bit to do.

Reply:
Ah, it looks like we check if the passed filter name is one of the serializer fields here:

```python
    def _get_field_or_error(self, field_name):
        """
        Check that the attempted filter field is valid

        :raises InvalidFilterError: If the filter field is not valid
        """
        serializer_class = self.serializer_class
        if field_name not in serializer_class._declared_fields:
            raise InvalidFilterError(detail="'{0}' is not a valid field for this endpoint.".format(field_name))
        if field_name not in getattr(serializer_class, 'filterable_fields', set()):
            raise InvalidFilterFieldError(parameter='filter', value=field_name)
        return serializer_class._declared_fields[field_name]
```

Does Django support hidden fields? Maybe if `read_only=True` and `write_only=True`?

Reply:
No need for this to block merging this PR, but it's worth a few minutes to see if we can avoid adding the extra field.